bin/check_fasta_insdc.py

```diff
@@ -64,29 +69,109 @@ def get_refget_metadata(md5_digest):
+    """
     results = {'sequences': []}
     all_insdc = True
+    possible_assemblies = set()
     for header, sequence in fasta_iter(input_fasta):
         name = header.split()[0]
         md5_digest = refget_md5_digest(sequence)
         sequence_metadata = get_refget_metadata(md5_digest)
-        results['sequences'].append({'sequence_name': name, 'sequence_md5': md5_digest, 'insdc': bool(sequence_metadata)})
-        all_insdc = all_insdc and bool(sequence_metadata)
+        is_insdc = bool(sequence_metadata)
+        if is_insdc:
+            containing_assemblies = get_containing_assemblies(md5_digest)
+            if len(containing_assemblies) == 0:
+                logger.warning(f'Sequence with this MD5 is INSDC but not found in contig alias: {md5_digest}')
+                continue
+            if len(possible_assemblies) == 0:
+                possible_assemblies = containing_assemblies
+            else:
+                possible_assemblies &= containing_assemblies
+        results['sequences'].append({'sequence_name': name, 'sequence_md5': md5_digest, 'insdc': is_insdc})
+        all_insdc = all_insdc and is_insdc
     results['all_insdc'] = all_insdc
+
+    # Only report on metadata concordance if all of the following hold:
+    #  1) All sequences in FASTA file are INSDC
+    #  2) At least one compatible assembly accession was found in contig alias
+    #  3) Found a single assembly accession in the metadata to compare against this FASTA file
+    # If (3) is missing but (1) and (2) hold, we will still report possible INSDC assemblies.
+    if all_insdc and possible_assemblies:
+        results['possible_assemblies'] = possible_assemblies
+    if all_insdc and possible_assemblies and metadata_insdc:
+        results['metadata_assembly_compatible'] = (metadata_insdc in possible_assemblies)
+        results['associated_analyses'] = analyses
+        results['assembly_in_metadata'] = metadata_insdc
     return results
 
 
+def get_analyses_and_reference_genome_from_metadata(vcf_files, json_file):
+    with open(json_file) as open_json:
+        metadata = json.load(open_json)
+        files_per_analysis = get_files_per_analysis(metadata)
+        # Get all analyses associated with all vcf files
+        all_analyses = set()
+        for vcf_file in vcf_files:
+            analysis_aliases = get_analysis_for_vcf_file(vcf_file, files_per_analysis)
+            if len(analysis_aliases) != 1:
+                logger.error(f'Could not determine analysis associated with VCF file: {vcf_file}')
+            else:
+                all_analyses.add(analysis_aliases[0])
+        # Get (single) assembly associated with all analyses
+        assemblies = [get_reference_assembly_for_analysis(metadata, analysis) for analysis in all_analyses]
+        if len(assemblies) != 1:
+            logger.error(f'Could not determine assembly accession to check against fasta file, out of: {assemblies}')
+            return all_analyses, None
```

Comment:
I think we should allow for multiple assemblies considering that it is possible to have different assemblies in different analysis.

Reply:
This should be not all the analyses, but all the analyses associated with VCF files that have been paired with this FASTA file in the mapping - via this [groupBy](https://github.com/EBIvariation/eva-sub-cli/blob/8046387fed1c45ed8549c6170f345c71f61982c2/eva_sub_cli/nextflow/validation.nf#L93) in the nextflow.
Unless you're talking about a case where there are different assembly accessions associated with the _same_ FASTA file? It's true that this doesn't allow for that, but I didn't think it could happen...

Reply:
I'll update the comments and the command line help text, they are definitely misleading here